trusted-types/should-trusted-type-policy-creation-be-blocked-by-csp-002.html

```diff
@@ -0,0 +1,78 @@
+<!DOCTYPE html>
+<meta charset="UTF-8">
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="./support/csp-violations.js"></script>
+<script>
+  const validTrustedTypePolicyNames = [
+    "1",
+    "abcdefghijklmnopqrstuvwxyz",
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
+    "0123456789",
+    "policy-name",
+    "policy=name",
+    "policy_name",
+    "policy/name",
+    "policy@name",
+    "policy.name",
+    "policy%name",
+    "policy#name",
+    "6xY/2x=3Y..."
+  ];
+  validTrustedTypePolicyNames.forEach(trustedTypePolicyName => {
+    promise_test(async t => {
+      let results = await tryCreatingTrustedTypePoliciesWithCSP(
+        [trustedTypePolicyName],
+        `header(Content-Security-Policy,trusted-types ${encodeURIComponent(trustedTypePolicyName)},True)`
+      );
+      assert_equals(results.length, 1);
+      assert_equals(results[0].name, trustedTypePolicyName);
+      assert_equals(results[0].exception, null);
+      assert_equals(results[0].violatedPolicies.length, 0);
+    }, `valid tt-policy-name name "${trustedTypePolicyName}"`);
+  });
+
+  const invalidTrustedTypePolicyNames = [
+    "policy name",
+    "policy$name", // https://github.com/w3c/trusted-types/issues/504
+    "policy?name",
+    "policy!name",
+    "policy*name",
+    "política",
+  ];
+  invalidTrustedTypePolicyNames.forEach(trustedTypePolicyName => {
+    promise_test(async t => {
+      let results = await tryCreatingTrustedTypePoliciesWithCSP(
+        [trustedTypePolicyName],
+        `header(Content-Security-Policy,trusted-types ${encodeURIComponent(trustedTypePolicyName)},True)`
+      );
+      assert_equals(results.length, 1);
+      assert_equals(results[0].name, trustedTypePolicyName);
+      assert_true(results[0].exception instanceof TypeError);
+      assert_equals(results[0].violatedPolicies.length, 1);
+    }, `invalid tt-policy-name name "${trustedTypePolicyName}"`);
+  });
+
+  promise_test(async t => {
+    let results = await tryCreatingTrustedTypePoliciesWithCSP(
+      ["_TTP1_", "_TTP2_", "_TTP3_", "_TTP4_", "_TTP5_", "_TTP6_"],
```

Comment:
All browsers fail on the _TTP3_ case. Can you explain what this bit is testing and why that is expected?

Reply:
Yeah, I'm not sure why they are failing. 

I updated the test to explain this is testing https://w3c.github.io/webappsec-csp/#grammardef-required-ascii-whitespace